### ISSUE TITLE
Croatia has no data, Slovenia lacks SI-HR data, but both is available.

### ISSUE THREAD

**Issue by seba1337** [tmrowco/electricitymap-contrib#1950]
I've found sources for both.

SI-HR data is available here https://www.eles.si/trzni-podatki, just seems you would need some work.

But even better, there is HR data, https://www.hops.hr/Home/PowerExchange in JSON, which gives also the missing SI-HR data and other interconnect data (SI = Slovenija, HU = Mađarska, RS+ME = Srbija i Crna Gora, BA = Bosna i Hercegovina) as well as total electricity production (Ukupna proizvodnja) and amount of electricity production by wind power (Proizvodnja VE).

I've found an old issue https://github.com/tmrowco/electricitymap-contrib/issues/1007, but seems that it wasn't realised that there is JSON data available, so no need for a complex parser. 

Citing wikipedia:
> Total installed capacity of generating objects built in Croatia amounts to 3745 MW, 2079 MW of which is hydropower's share, and 1666 MW comes from thermal power plants. There are 25 hydroelectric power plants, most of which are located on the coast (including the single pumped storage hydroelectric plant, RHE Velebit), and 7 thermal power stations, 3 of which also generate heat for industry and heating in cities. Also, 338 MW is available from the co-owned Krško Nuclear Power Plant, and 210 MW from the privately owned Plomin 2 thermal power plant

Seems that that's old data. Not having wind power, but we also see that a large share of installed power is hydro power of which there's no(?) public data. :'(

Croatian wikipedia has fresher data, there seems 570 MW wind power (2018), source: https://hr.wikipedia.org/wiki/Vjetroelektrane_u_Hrvatskoj. 

Seeing this site about alternative energy, http://www.poslovni.hr/hrvatska/u-hrvatskoj-je-u-pogonu-1207-elektrana-na-oie-koliko-nam-je-preostalo-do-ispunjenja-zadanih-ciljeva-do-2020-300342#, seems 40,4 MW solar, 15,9 MW biogas, 14,9 MW biomass, 5,5 MW landfill and waste water gas, 0 MW geothermal. The data on hydro is on microhydro, so not relevant. 

The data on hydroplants in the first citation (2079 MW) seems approx. correct.

Thermal power plants changed too,
* 303 MW oil (TE Rijeka, is in reserve, but used in peaks) 
* 199 MW coal (TE Plomin, block A is taken offline)
* 78 MW gas (KTE Jertovec)
* 300 MW gas (TE-TO Zagreb)
* 50 MW gas (EL-TO Zagreb)
* 89 MW gas (TE-TO Osijek)
* 229 MW gas (TE-TO Sisak)

So in total 746 MW gas, 199 MW coal and 303 MW oil = 1248 MW thermo, a bit less than in the first citation.

There seems to be also around 1 MW of pumping storage water plants.

**Comment by scriptator** [comment]
Maybe installed installed capacities can be taken from the corresponding [ENTSOE page](https://transparency.entsoe.eu/generation/r2/installedGenerationCapacityAggregation/show?name=&defaultValue=true&viewType=TABLE&areaType=BZN&atch=false&dateTime.dateTime=01.01.2019+00:00|UTC|YEAR&dateTime.endDateTime=01.01.2019+00:00|UTC|YEAR&area.values=CTY|10YHR-HEP------M!BZN|10YHR-HEP------M&productionType.values=B01&productionType.values=B02&productionType.values=B03&productionType.values=B04&productionType.values=B05&productionType.values=B06&productionType.values=B07&productionType.values=B08&productionType.values=B09&productionType.values=B10&productionType.values=B11&productionType.values=B12&productionType.values=B13&productionType.values=B14&productionType.values=B20&productionType.values=B15&productionType.values=B16&productionType.values=B17&productionType.values=B18&productionType.values=B19). Do the values there look sensible to you?

**Comment by systemcatch** [comment]
Nice research @seba1337, I wrote the old parser (https://github.com/tmrowco/electricitymap-contrib/blob/master/parsers/HOPS.py) for HR which only gets 2 exchanges. They've clearly updated the website significantly, we might even be to have a real-time parser (it'll show unknown/wind only) which precise backfilling from ENTSOE.

**Comment by seba1337** [comment]
> 
> 
> Maybe installed installed capacities can be taken from the corresponding [ENTSOE page](https://transparency.entsoe.eu/generation/r2/installedGenerationCapacityAggregation/show?name=&defaultValue=true&viewType=TABLE&areaType=BZN&atch=false&dateTime.dateTime=01.01.2019+00:00%7CUTC%7CYEAR&dateTime.endDateTime=01.01.2019+00:00%7CUTC%7CYEAR&area.values=CTY%7C10YHR-HEP------M!BZN%7C10YHR-HEP------M&productionType.values=B01&productionType.values=B02&productionType.values=B03&productionType.values=B04&productionType.values=B05&productionType.values=B06&productionType.values=B07&productionType.values=B08&productionType.values=B09&productionType.values=B10&productionType.values=B11&productionType.values=B12&productionType.values=B13&productionType.values=B14&productionType.values=B20&productionType.values=B15&productionType.values=B16&productionType.values=B17&productionType.values=B18&productionType.values=B19). Do the values there look sensible to you?

There seems to be way too much thermo. Those TE/EL-TO thermo plants (CHP actually) can run either on gas or extra light fuel oil. It seems like they double counted them. For gas the sums agree, but for fossil oil is like they counted gas fired ones again + added a 194 MW plant. Fossil coal I think is old data, the Plomin block A is shut down only recently, it seems like they count Plomin block A+B. Not sure what 61 MW other renewables would be? Wind could be that is so high, I have 2018 data and they are heavily investing in wind. Also pumped hydro seems suspiciously high, would have to check.

TLDR, I think they double counted TE-TO plants (in both gas and fuel oil) because they can use gas or light fuel oil.

**Comment by seba1337** [comment]
> 
> 
> Nice research @seba1337, I wrote the old parser (https://github.com/tmrowco/electricitymap-contrib/blob/master/parsers/HOPS.py) for HR which only gets 2 exchanges. They've clearly updated the website significantly, we might even be to have a real-time parser (it'll show unknown/wind only) which precise backfilling from ENTSOE.

I actually just reverse engineered their site, i.e. analyzed traffic on the front page and found the JSON. :-) I haven't found anywhere that they've published the JSON. It's real time, but shows only what I've written, but at least something. 

Anyway, I speak both Slovenian and Croatian, if something won't be understandable and google translate won't save the day.

**Comment by PaulCornelissen** [comment]
So I had a little look at this and sofar I have this:
```
                sourceName  status     value
0      Žerjavinec - Heviz1       0   251.000
1      Žerjavinec - Heviz2       0     0.000
2          Tumbri - Krško1       0   232.000
3          Tumbri - Krško2       0   234.000
4      Ernestinovo - Pecs1       0   246.000
5      Ernestinovo - Pecs2       0     0.000
6          Melina - Divača       0   105.000
7         Konjsko - Mostar       0   156.000
8   Ernestinovo - S. Mit.2       0     0.000
9   Ernestinovo - Ugljevik       0   -35.000 ~= Coal power plant in BA https://en.wikipedia.org/wiki/Ugljevik_Power_Plant
10               Slovenija       0   789.000 == Exchange Slovenia
11                Mađarska       0   541.000 == Exchange Hungary
12      Srbija i Crna Gora       0   -12.000 == Exchange Serbia and Montenegro
13     Bosna i Hercegovina       0   214.000 == Exchange with Bonia and Herzegovina
14             Frekvencija       0    50.021 == Grid frequency
15       Opterećenje EES-a       0  2299.000 ~= EES Load
16      Ukupna proizvodnja       0   798.000 == Total generation
17          Proizvodnja VE       0    31.000 == Wind generation
```
The first 9 rows to me appear to be power plants, but I'm not really able to attribute them. The sum of their production (assuming it's live) doesn't add up to total production mentioned under total generation. If these are plants in Croatia and we know the fuel type, then we might be able to attribute a large part of the power production to a source and give it CO2 intensity.

Also I found that the 4 exchanges with neighboring countries are being published via ENTSOE. So I changed the exchange mapping and that works. I'll make a PR for that at least, so we can at least see all exchanges on the map.

**Comment by seba1337** [comment]
The first **ten (10)** are exchanges (not 9), i.e. various high voltage lines. For example Tumbri - Krško1 is one of the high voltage lines connectin Slovenia and Croatia. Ernestinovo - Ugljevik is also such an exchange and not a power plant.

EES I think is "elektro-energetski sustav", electro-energetic system. The rest seems to be alright.
In this case you could do:

Total generation: 798 MW
Wind: 31 MW
Unknown: 798 MW - 31 MW = 767 MW.

**Comment by systemcatch** [comment]
@PaulCornelissen all done here?

**Comment by PaulCornelissen** [comment]
@systemcatch It seems to work. 

We could still improve slightly if someone figured out a more accurate CO2 intensity for unknown production in HR.

Also I simply copied the capacities from ENTSOE, which as discussed above, are probably to high for some technologies. Isn't really a big issue I'd say.

**Comment by systemcatch** [comment]
@PaulCornelissen ok then let's close this for now and reopen if needed in the future.

**Comment by seba1337** [comment]
> 
> 
> @systemcatch It seems to work.
> 
> We could still improve slightly if someone figured out a more accurate CO2 intensity for unknown production in HR.
> 
> Also I simply copied the capacities from ENTSOE, which as discussed above, are probably to high for some technologies. Isn't really a big issue I'd say.

https://www.hops.hr/page-file/oEvvKj779KAhmQg10Gezt2/temeljni-podaci/Temeljni%20podaci%202018.pdf

For 2018 it's 6691 GWh hydro, 3206 GWh thermo, 1240 GWh wind. So unknown is 6691 GWh + 3206 GWh = 9897 GWh. Hydro taking 68% of the unknown.
